modules/features/sharing/src/debug/kotlin/au/com/shiftyjelly/pocketcasts/sharing/FFmpegMediaService.kt

```diff
@@ -0,0 +1,91 @@
+import au.com.shiftyjelly.pocketcasts.sharing.clip.Clip
+import au.com.shiftyjelly.pocketcasts.utils.toHhMmSs
+import com.arthenica.ffmpegkit.FFmpegKit
+import com.arthenica.ffmpegkit.FFmpegSession
+import com.arthenica.ffmpegkit.FFmpegSessionCompleteCallback
+import com.arthenica.ffmpegkit.ReturnCode
+import java.io.File
+import java.io.IOException
+import kotlin.coroutines.resume
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+
+internal class FFmpegMediaService(
+    private val context: Context,
+) : MediaService {
+    override suspend fun clipAudio(podcast: Podcast, episode: PodcastEpisode, clipRange: Clip.Range): Result<File> = withContext(Dispatchers.IO) {
+        val outputFile = File(context.cacheDir, "${podcast.title} - ${episode.title} - ${clipRange.start.toHhMmSs()}–${clipRange.end.toHhMmSs()}.mp3")
+        if (outputFile.exists()) {
+            return@withContext Result.success(outputFile)
+        }
+
+        val ffmpegFile = File(context.cacheDir, "ffmpeg-audio-clip.mp3")
+        val command = buildString {
+            val audioSource = episode.downloadedFilePath?.let(::File)?.takeIf(File::exists)?.toString() ?: episode.downloadUrl
+            if (audioSource == null) {
+                return@withContext Result.failure(IOException("No stream found for ${episode.uuid} episode"))
+            }
+
+            append("-i $audioSource ") // Audio stream source
+            val coverFile = convertCoverToJpeg(episode).getOrNull() // Convert covers to JPEG because MP3 doesn't support embedding WebP
+            if (coverFile != null) {
+                append("-i $coverFile ")
+            }
+            append("-ss ${clipRange.startInSeconds} ") // Clip start, must be in seconds or HH:MM:SS.xxx
+            append("-to ${clipRange.endInSeconds} ") // Clip end, must be in seconds or HH:MM:SS.xxx
+            append("-q:a 0 ") // Max audio quality
+            append("-map 0:a:0 ") // Include only the first audio stream, videos can have multiple audio streams
+            if (coverFile != null) {
+                append("-map 1:0 ") // Include the cover stream
+                append("-c:1 copy ") // Copy codec for the cover stream
+            }
+            append("-y ") // Overwrite output file if it already exists
+            append("$ffmpegFile") // Output file
+        }
+
+        executeAsyncCommand(command).mapCatching {
+            if (!ffmpegFile.renameTo(outputFile)) {
+                throw IOException("Failed to rename clip file to output file")
+            }
+            outputFile
+        }
+    }
+
+    private suspend fun convertCoverToJpeg(episode: PodcastEpisode): Result<File> {
+        val outputFile = File(context.cacheDir, "ffmpeg-converted-cover.jpeg")
+        val coverPath = episode.imageUrl ?: "${BuildConfig.SERVER_STATIC_URL}/discover/images/webp/960/${episode.podcastUuid}.webp"
```

Comment:
We do have a .jpg podcast image if that helps, rather than having the download the .webp.
```
https://static.pocketcasts.com/discover/images/960/4eb5b260-c933-0134-10da-25324e2a541d.jpg
```

Reply:
I was sure we had one but I was looking for it with `/images/jpg/` path. I'll change though I'll keep this conversion anyway since we don't know what is behind `imageUrl`.

Reply:
Fixed in https://github.com/Automattic/pocket-casts-android/pull/2545/commits/6515d4cfd9486b4e34670b4d86ce7cd332d8c8b1